src/components/postDropdown/container/postDropdownContainer.tsx

```diff
@@ -177,11 +182,137 @@ class PostDropdownContainer extends PureComponent {
           },
         });
         break;
+      case 'mute':
+        this._handleMuteUnmuteUser(!isMuted);
+        break;
+      case 'unmute':
+        this._handleMuteUnmuteUser(!isMuted);
+        break;
       default:
         break;
     }
   };
 
+  _handleMuteUnmuteUser = async (isMuteAction) => {
+    const { currentAccount, content } = this.props as any;
+    const username = content.author;
+    const isOwnProfile = !username || currentAccount.username === username;
+    console.log('isMuteAction,isOwnProfile : ', isMuteAction, isOwnProfile);
+
+    if(!isOwnProfile){
+      console.log('isMuteAction : ', isMuteAction);
+      
+      if (isMuteAction) {
+        this._muteUser();
+      } else {
+        this._handleFollowUnfollowUser();
+      }
+    }
+    
+  };
+  _muteUser = () => {
+    const { currentAccount, pinCode, dispatch, intl, content, onLoadPosts } = this.props as any;
+    const username = content.author;
+    const follower = currentAccount.name;
+    const following = username;
+   
+    ignoreUser(currentAccount, pinCode, {
+      follower,
+      following,
+    })
+      .then(() => {
+        const curMutes = currentAccount.mutes || [];
+        if (curMutes.indexOf(username) < 0) {
+          //check to avoid double entry corner case
+          currentAccount.mutes = [username, ...curMutes];
+        }
+        dispatch(updateCurrentAccount(currentAccount));
+        onLoadPosts(true); //fetch posts again after muting user
+        dispatch(
+          toastNotification(
+            intl.formatMessage({
+              id: 'alert.success_mute',
+            }),
+          ),
+        );
+      })
+      .catch((err) => {
+        this._profileActionDone({ error: err });
+      });
+ 
+  };
+
+  _handleFollowUnfollowUser = async () => {
```

Comment:
@aliseyalvi do we really need this considering, I think we can just keep feed limited to blocking. user my unblock later from profile, also we can give option somewhere in profile to view a list of blocked users and unblock them if required.

Reply:
I think if post is showing in screen but content is hidden, it would be easy for user to unblock again. But we can remove that , what u suggest